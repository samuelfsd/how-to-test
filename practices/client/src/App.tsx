export function App() {
  return (
    <>
      <h1 className="text-6xl text-red-500">Ol√° mundo!</h1>
    </>
  )
}

