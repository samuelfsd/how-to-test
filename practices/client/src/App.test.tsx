// Não teste implementação e sim teste comportamento !!!!

import { render, screen } from '@testing-library/react'
import { describe, expect, it } from  'vitest'
import { App } from './App'

describe('Render App', () => {
    it('should be render App component', () => {
        const { container } = render(<App />)

        const headingElement = screen.getByRole('heading', { name: 'Olá mundo!' })

        expect(container).toMatchSnapshot();
        expect(headingElement).toBeDefined();
    })
})