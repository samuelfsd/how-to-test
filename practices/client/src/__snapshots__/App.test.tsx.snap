// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render App > should be render App component 1`] = `
<div>
  <h1>
    Ol√° mundo!
  </h1>
</div>
`;
